custom-plugin-rules/markdown-validator/rule-validate-markdown.js

```diff
@@ -13,32 +13,32 @@ function checkString(description, ctx) {
     },
     config: config,
   };
-  markdownlint(options, function callback(err, result) {
-    if (!err) {
-      // if there's no problem do nothing
-      if (result.desc.length) {
-        // desc is the key in the options.strings object
-        let lines = description.split("\n");
 
-        result.desc.forEach((desc) => {
-          message = desc.ruleDescription;
-          // add line number context for longer entries
-          if (desc.lineNumber > 1) {
-            message =
-              message +
-              " (near: " +
-              lines[desc.lineNumber].substring(0, 20) +
-              "... )";
-          }
+  try {
+    const lintResults = markdownlint.sync(options);
+    
+    if (lintResults.desc.length) {
+      // desc is the key in the options.strings object
+      let lines = description.split("\n");
 
-          ctx.report({
-            message: message,
-            location: ctx.location.child("description"),
-          });
+      lintResults.desc.forEach((desc) => {
```

Comment:
Could you rewrite this to `for` ... `of` instead of `forEach`? It should be a little bit better in terms of readability.